workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/SpatialOSBehaviourLibrary.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections.Generic;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    /// <summary>
+    /// Retrieves Reader and Writer fields from MonoBehaviours and handles injection into them.
+    /// </summary>
+    public class SpatialOSBehaviourLibrary {
+        // Stores maps from Spatial component IDs to MemberAdapters to reflectively set Readers/Writers for each SpatialOSBehaviour.
+        private readonly Dictionary<Type, Dictionary<uint, IMemberAdapter>> adapterCache
+            = new Dictionary<Type, Dictionary<uint, IMemberAdapter>>();
+
+        // List of component IDs for which a reader/writer is required for each SpatialOSBehaviour
+        private readonly Dictionary<Type, List<uint>> componentReaderIdsForBehaviours = new Dictionary<Type, List<uint>>();
+        private readonly Dictionary<Type, List<uint>> componentWriterIdsForBehaviours = new Dictionary<Type, List<uint>>();
+
+        // Helper objects for querying members with the [Require] attribute
+        private readonly MemberReflectionUtil reflectionUtil = new MemberReflectionUtil(typeof(RequireAttribute));
+
+        private readonly HashSet<Type> badTypes = new HashSet<Type>();
+
+        private readonly ILogDispatcher logger;
+        private const string LoggerName = "SpatialOSBehaviourLibrary";
+
+        private const string BadRequiredMemberWarning
+            = "[Require] attribute found on member that is not Reader or Writer or incorrectly generated, ignoring!";
+
+        private const string MultipleReadersWritersRequiredError
+            = "MonoBehaviour found requesting more than one view (Reader or Writer) to the same component, " +
```

Comment:
Don't mention the word "view" here as that might confuse people. Just say `MonoBehaviour found requesting more than one Reader or Writer for the same component...`

Reply:
I was trying to figure out how to communicate that I'm talking about more than one [Reader or Writer], rather than [more than one reader] or [more than one writer], ie you can't even have a reader and a writer. Any suggestions on how to make that clear?

Reply:
I'd go with Jonas's suggestion. View is an overloaded term (and not quite right here).